lib/cisco_node_utils/snmp_notification_receiver.rb

```diff
@@ -0,0 +1,158 @@
+                   source_interface: '')
+
+      fail TypeError unless name.is_a?(String)
+      @name = name
+
+      fail TypeError unless type.is_a?(String)
+
+      fail TypeError unless version.is_a?(String) || version.is_a?(Integer)
+
+      fail TypeError unless security.is_a?(String)
+
+      fail TypeError unless username.is_a?(String)
+
+      fail TypeError unless port.is_a?(String) || version.is_a?(Integer)
+
+      fail TypeError unless vrf.is_a?(String)
+
+      fail TypeError unless source_interface.is_a?(String)
+
+      return unless instantiate
+
+      # Mandatory Properties
+      fail TypeError unless name.length > 0
+      fail TypeError unless type.length > 0
+      fail TypeError unless version.length > 0
+      fail TypeError unless username.length > 0
+
+      config_set('snmp_notification_receiver',
+                 'receivers',
+                 state:    '',
+                 ip:       name,
+                 type:     type,
+                 version:  version,
+                 security: security,
+                 username: username,
+                 udp_port: port.empty? ? '' : "udp-port #{port}")
+
+      unless source_interface.empty?
+        config_set('snmp_notification_receiver',
+                   'source_interface',
+                   ip:               name,
+                   source_interface: source_interface,
+                   port:             port.empty? ? '' : "udp-port #{port}")
+      end
+
+      return if vrf.empty?
+      config_set('snmp_notification_receiver',
+                 'vrf',
+                 ip:   name,
+                 vrf:  vrf,
+                 port: port.empty? ? '' : "udp-port #{port}")
+    end
+
+    def self.receivers
+      hash = {}
+
+      receivers_list = config_get('snmp_notification_receiver', 'receivers')
+      return hash if receivers_list.nil?
+
+      receivers_list.each do |arr|
+        next if !arr.is_a?(Array) || arr.empty?
+        id = arr[0]
+        hash[id] = SnmpNotificationReceiver.new(id, instantiate: false)
+      end
+
+      hash
+    end
+
+    def ==(other)
+      name == other.name
+    end
+
+    def destroy
+      config_set('snmp_notification_receiver',
+                 'receivers',
+                 state:    'no',
+                 ip:       name,
+                 type:     type,
+                 version:  version,
+                 security: security.nil? ? '' : "#{security}",
+                 username: username,
+                 udp_port: port.nil? ? '' : "udp-port #{port}")
+    end
+
+    def port
+      val = config_get('snmp_notification_receiver', 'port', @name)
+      val
+    end
+
+    def username
+      if !port.nil?
+        endpoint = 'username_with_port'
+      else
+        endpoint = 'username'
+      end
+
+      val = config_get('snmp_notification_receiver', endpoint, @name)
+      val
+    end
+
+    def version
+      val = config_get('snmp_notification_receiver', 'version', @name)
+      val
```

Comment:
Assign and return is unnecessary in all of these getters - they can be rewritten more succinctly:

```ruby
def version
  config_get('snmp_notification_receiver', 'version', @name)
end
```